charms/jupyter-ui/tests/unit/test_operator.py

```diff
@@ -29,14 +38,34 @@ def harness() -> Harness:
 class TestCharm:
     """Test class for JupyterUI."""
 
-    def test_spawner_ui(self):
+    def render_spawner_ui_config(self, jupyter_images, vscode_images, rstudio_images):
+        """Renders the spawner config template with the config values."""
+        environment = Environment(loader=FileSystemLoader("."))
+        template = environment.get_template(JWA_CONFIG_FILE)
+        yaml_content = template.render(
+            jupyter_images=jupyter_images,
+            vscode_images=vscode_images,
+            rstudio_images=rstudio_images,
+        )
+        content = yaml.safe_load(yaml_content)
+        return content
+
+    def test_spawner_ui(self, harness):
         """Test spawner UI.
 
-        spawner_ui_config.yaml contains a number of changes that were done for Charmed
-        Kubeflow. This test is to validate those. If it fails, spawner_ui_config.yaml
+        spawner_ui_config.yaml.j2 contains a number of changes that were done for Charmed
+        Kubeflow. This test is to validate those. If it fails, spawner_ui_config.yaml.j2
         should be reviewed and changes to this tests should be made, if required.
         """
-        spawner_ui_config = yaml.safe_load(Path("./src/spawner_ui_config.yaml").read_text())
+        # Load the default config for Notebook images lists
+        jupyter_images = yaml.safe_load(harness.model.config[JUPYTER_IMAGES_CONFIG])
+        vscode_images = yaml.safe_load(harness.model.config[VSCODE_IMAGES_CONFIG])
+        rstudio_images = yaml.safe_load(harness.model.config[RSTUDIO_IMAGES_CONFIG])
+
+        # Render the spawner config file with the default configs
+        spawner_ui_config = self.render_spawner_ui_config(
+            jupyter_images, vscode_images, rstudio_images
+        )
```

Comment:
In this test, we're kind of testing the default config by mocking the actual process of going from juju config to rendered configmap.  Is that what you want to test?  

I wonder if there's a way you could be using the functionality in your charm code or helpers to do this so we could confirm *the charm* knew how to go from juju config to configmap?